src/main/java/conversion/utils/HttpHelper.java

```diff
@@ -0,0 +1,109 @@
+ * limitations under the License.
+ *
+ */
+package conversion.utils;
+
+import conversion.Individual;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Static collection of methods to help with sending http requests.
+ */
+public class HttpHelper {
+
+    private static Logger LOG = Logger.getLogger(HttpHelper.class.getName());
+    private static int MaxRetries = 1;
+
+    /**
+     * Tries to send individual json data to the callbackUrl provided by the
+     * user when the file is submitted.
+     *
+     * @param callbackUrl
+     * @param individual
+     * @return HTTP response code
+     * @throws MalformedURLException
+     * @throws IOException
+     */
+    public static int contactCallback(String callbackUrl, Individual individual) throws MalformedURLException, IOException {
+        final URL callback = new URL(callbackUrl);
+
+        final HttpURLConnection connection = (HttpURLConnection) callback.openConnection();
+
+        connection.setRequestMethod("POST");
+        connection.setDoOutput(true);
+        connection.setRequestProperty("charset", "utf-8");
+        connection.setRequestProperty("Content-Type", "application/json");
+        connection.setRequestProperty("Accept", "application/json");
+
+        final String outputString = individual.toJsonString();
+
+        final byte[] outputBytes = outputString.getBytes("UTF-8");
+        connection.setFixedLengthStreamingMode(outputBytes.length);
+
+        try (OutputStream os = connection.getOutputStream(); OutputStreamWriter wr = new OutputStreamWriter(os)) {
+            wr.write(outputString);
+            wr.flush();
+            wr.close();
+            os.close();
+        }
+
+        return connection.getResponseCode();
+    }
+
+    /**
+     * This method handles the process of sending the callback data and handles
+     * any failed attempts. It will retry once every 10 seconds until it reaches
+     * the MaxRetries, this value is set to 1 by default.
+     *
+     * @param callbackUrl The URL which will receive the json data
+     * @param individual The data to be sent to the callbackUrl
+     */
+    public static void sendCallback(String callbackUrl, Individual individual) {
+        try {
+            int resCode = contactCallback(callbackUrl, individual);
+            int currentRetries = 1;
+
+            while (resCode != HttpURLConnection.HTTP_OK && !(currentRetries > MaxRetries)) {
+                LOG.log(Level.WARNING, "Callback URL ''{0}'' returned http code: {1} on attempt no.{2}", new Object[]{callbackUrl, Integer.toString(resCode), currentRetries});
+
+                try {
+                    Thread.sleep(10000);
+                } catch (InterruptedException e) {
+                    LOG.info("Callback Retry interupted");
```

Comment:
I would use LOG.severe here

Also interrupted is missing an r

Reply:
Thread.sleep is no longer used, but good spot